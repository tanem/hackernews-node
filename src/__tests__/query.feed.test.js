const path = require('path')
const bcrypt = require('bcryptjs')
const chance = require('chance').Chance(path.basename(__filename))
const request = require('supertest')

let httpServerUrl
let prisma

beforeAll(async () => {
  httpServerUrl = global.__HTTP_SERVER_URL__
  prisma = global.__PRISMA__
})

const name = chance.first()
const email = chance.email()
const password = chance.word()
const links = new Array(5).fill().map(() => ({
  description: chance.sentence({ words: 5 }),
  url: chance.url(),
}))

beforeEach(async () => {
  await prisma.user.create({
    data: {
      name,
      email,
      password: await bcrypt.hash(password, 10),
      links: {
        create: links,
      },
    },
  })
})

test('simple', async () => {
  const { body, status } = await request(httpServerUrl).post('/').send({
    query: `
      query {
        feed {
          count
          links {
            id
            description
            url
            postedBy {
              id
              name
            }
            votes {
              id
              user {
                id
                name
              }
            }
          }
        }
      }
    `,
  })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot()
})

test('filter', async () => {
  const { body, status } = await request(httpServerUrl).post('/').send({
    query: `
      query {
        feed(filter: "mo") {
          count
          links {
            id
            description
            url
            postedBy {
              id
              name
            }
            votes {
              id
              user {
                id
                name
              }
            }
          }
        }
      }
    `,
  })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot()
})

test('pagination', async () => {
  const { body, status } = await request(httpServerUrl).post('/').send({
    query: `
      query {
        feed(
          take: 3
          skip: 1
        ) {
          count
          links {
            id
            description
            url
            postedBy {
              id
              name
            }
            votes {
              id
              user {
                id
                name
              }
            }
          }
        }
      }
    `,
  })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot()
})

test('sorting', async () => {
  const { body, status } = await request(httpServerUrl).post('/').send({
    query: `
      query {
        feed(orderBy: { description: asc }) {
          count
          links {
            id
            description
            url
            postedBy {
              id
              name
            }
          }
        }
      }
    `,
  })

  expect(status).toBe(200)
  expect(body).toMatchSnapshot()
})
