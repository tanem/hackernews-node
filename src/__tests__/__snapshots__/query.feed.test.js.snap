// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "count": 2,
      "links": Array [
        Object {
          "description": "Vehis gupjunomo tafkizhib kifka vurme.",
          "id": "1",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://digikul.dj/fof",
          "votes": Array [],
        },
        Object {
          "description": "Sapuja bit aveanahar wobibimo wis.",
          "id": "2",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://gepinu.ug/bodzud",
          "votes": Array [],
        },
      ],
    },
  },
}
`;

exports[`pagination 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "count": 5,
      "links": Array [
        Object {
          "description": "Sapuja bit aveanahar wobibimo wis.",
          "id": "2",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://gepinu.ug/bodzud",
          "votes": Array [],
        },
        Object {
          "description": "Bewo dakoslih da ih gid.",
          "id": "3",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://lipbirgu.sy/ija",
          "votes": Array [],
        },
        Object {
          "description": "Nab ruun mu suolpih dutied.",
          "id": "4",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://jehne.kz/zenok",
          "votes": Array [],
        },
      ],
    },
  },
}
`;

exports[`simple 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "count": 5,
      "links": Array [
        Object {
          "description": "Vehis gupjunomo tafkizhib kifka vurme.",
          "id": "1",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://digikul.dj/fof",
          "votes": Array [],
        },
        Object {
          "description": "Sapuja bit aveanahar wobibimo wis.",
          "id": "2",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://gepinu.ug/bodzud",
          "votes": Array [],
        },
        Object {
          "description": "Bewo dakoslih da ih gid.",
          "id": "3",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://lipbirgu.sy/ija",
          "votes": Array [],
        },
        Object {
          "description": "Nab ruun mu suolpih dutied.",
          "id": "4",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://jehne.kz/zenok",
          "votes": Array [],
        },
        Object {
          "description": "Mer tewozme pug luz lesset.",
          "id": "5",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://nifoh.ee/ku",
          "votes": Array [],
        },
      ],
    },
  },
}
`;

exports[`sorting 1`] = `
Object {
  "data": Object {
    "feed": Object {
      "count": 5,
      "links": Array [
        Object {
          "description": "Bewo dakoslih da ih gid.",
          "id": "3",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://lipbirgu.sy/ija",
        },
        Object {
          "description": "Mer tewozme pug luz lesset.",
          "id": "5",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://nifoh.ee/ku",
        },
        Object {
          "description": "Nab ruun mu suolpih dutied.",
          "id": "4",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://jehne.kz/zenok",
        },
        Object {
          "description": "Sapuja bit aveanahar wobibimo wis.",
          "id": "2",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://gepinu.ug/bodzud",
        },
        Object {
          "description": "Vehis gupjunomo tafkizhib kifka vurme.",
          "id": "1",
          "postedBy": Object {
            "id": "1",
            "name": "Essie",
          },
          "url": "http://digikul.dj/fof",
        },
      ],
    },
  },
}
`;
